packages/next/src/client/components/router-reducer/reducers/navigate-reducer.ts

```diff
@@ -248,20 +286,28 @@ export function navigateReducer(
       ? createHrefFromUrl(canonicalUrlOverride)
       : href
 
-    mutable.previousTree = currentTree
-    mutable.patchedTree = newTree
-    mutable.applyFocusAndScroll = true
-    mutable.pendingPush = pendingPush
-    mutable.hashFragment = hash
-
     const applied = applyFlightData(currentCache, cache, flightDataPath)
     if (applied) {
       mutable.cache = cache
       currentCache = cache
     }
 
     currentTree = newTree
+
+    for (const subSegment of generateSegmentsFromPatch(treePatch)) {
+      scrollableSegments.push(
+        [...flightSegmentPath, ...subSegment].filter(
+          (segment) => segment !== '__PAGE__'
```

Comment:
not too sure what's happening here, but have you thought of params? In my case i have `"__PAGE__?{"slug":["blog-post-title"]}"` and this causing an issue with `!focusAndScrollRef.segmentPaths.some` in layout-router evaluating to true - eventually not scrolling to the top of the page when navigating to a new page.

Reply:
It's a good catch! I have a task for it, I realised I messed up the other day as well

Reply:
im facing this issue only when running the production build - dev mode is fine somehow

Reply:
many thanks for the update 🙏🏻